### PR TITLE
Restructure tls protocols packages

### PR DESCRIPTION
**Pull request by val06**
Moving all files related to tls protocols (java, gotls, openssl) under protocols/tls directory and structuring them in sub-directories.

this includes also the testutils and testdata used for UTs

Motivation
Better structuring of code in the repository
### Describe how to test/QA your changes

Make sure `agent-usm.jar` exists in the final docker artifact

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/network/usm/ebpf_javatls.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ package usm
 import (
 	"bytes"
 	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/network/protocols/tls/java"
```
Comment (DanielLavie) [inline pkg/network/usm/ebpf_javatls.go:13]: Move to correct section

File: omnibus/config/software/system-probe.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@
   mkdir "#{install_dir}/embedded/share/system-probe/ebpf/co-re/btf"
   mkdir "#{install_dir}/embedded/share/system-probe/java"
 
-  copy 'pkg/network/java/agent-usm.jar', "#{install_dir}/embedded/share/system-probe/java/"
+  copy 'pkg/network/protocols/tls/java/agent-usm.jar', "#{install_dir}/embedded/share/system-probe/java/"
```
Comment (guyarb) [inline omnibus/config/software/system-probe.rb:20]: please add to the QA steps, to ensure `agent-usm.jar` exists in the final docker artifact

Comment (val06) [reply omnibus/config/software/system-probe.rb:20]: added to QA testing section

**Review by mbotarro — APPROVED** [review]
looks good for changes touching processes files.

**Review by DanielLavie — APPROVED** [review]

**Review by usamasaqib — APPROVED** [review]
Looks good for @DataDog/ebpf-platform owned files.

**Review by amenasria — APPROVED** [review]
LGTM for Agent Platform owned file !

**Review by hmahmood — APPROVED** [review]

**Pull request merged**